articles/k-means-from-scratch-R/index.md

```diff
@@ -0,0 +1,311 @@
+    [2,] 0.25 1.00
+    [3,] 4.00 9.00
+    [4,] 16.00 36.00
+    [5,] 6.25 16.00
+    [6,] 12.25 16.00
+    [7,] 6.25 12.25
+    ```
+3. Addition and Square root:
+
+    ```r
+    distance_from_cluster_1 = sqrt(distance_from_cluster_1[,1] + distance_from_cluster_1[,2])
+    distance_from_cluster_1
+    ```
+    ```txt
+    [1] 0.000000 1.118034 3.605551 7.211103 4.716991 5.315073 4.301163
+    ```
+
+This `distance_from_cluster_1` represents the distance between each point and the centroid-1. Similarly, we calculate the distances for centroid-2.
+
+```r
+distance_from_cluster_2 = (dataPoints[,] - cent[2,])^2
+distance_from_cluster_2 = sqrt(distance_from_cluster_2[,1] + distance_from_cluster_2[,2])
+distance_from_cluster_2
+```
+
+The values of `distance_from_cluster_2` is given as follows:
+
+```txt
+[1] 5.376453 4.419417 1.776584 2.899353 0.728869 1.237437 1.075291
+```
+
+Combining `distance_from_cluster_1` and `distance_from_cluster_2`, we get the total distance array called `total_distance.` This array contains the distances between points and the centroids. We use this array to determine which cluster a given point belongs to.
+
+```r
+total_distance = array(c(distance_from_cluster_1, distance_from_cluster_2), dim = c(7, 2))
+total_distance
+```
+
+The array `total_distance` is shown below:
+
+```txt
+ [,1] [,2]
+[1,] 0.000000 7.211103
+[2,] 1.118034 6.264982
+[3,] 3.605551 3.605551
+[4,] 7.211103 2.828427
+[5,] 4.716991 2.500000
+[6,] 5.315073 2.500000
+[7,] 4.301163 2.915476
+```
+
+#### Step 4: Compare and Find the Closest Centroids
+Let's create a logical vector comparing `distance_from_cluster_1` and `distance_from_cluster_2`. This vector will be comprised of the Boolean values `TRUE` and `FALSE.` Consider the example of creating this vector using a conditional statement.
+
+The condition would be as follows: distance to the first cluster is less than the second cluster's distance. Points that satisfy this condition belong to cluster 1. The remaining points belong to cluster 2.
+
+```r
+c(total_distance[,1] <= total_distance[,2])
+```
+```txt
+TRUE TRUE TRUE FALSE FALSE FALSE FALSE
+```
+
+Using the logical vector above, we obtain the elements of the first cluster. The operation used below is an example of conditional selection. Elements that satisfy the condition in the array `dataPoints` are printed.
+
+```r
+dataPoints[,1][c(total_distance[,1] <= total_distance[,2])]
+```
+
+```txt
+1.0 1.5 3.0
+```
+
+To find the centroid of the newly formed cluster, we take the mean of all the points obtained above.
+The thinking is as follows: We need to find a point closest to all the cluster data points. Therefore, averaging the data points results in a point closest to the remaining points.
+
+```r
+mean(dataPoints[,1][c(total_distance[,1] <= total_distance[,2])])
+```
+
+We calculate the mean using the R function `mean.` This is an example on selecting elements conditionally that belong to a cluster and finding its centroid.
```

Comment:
If we could clarify this sentence: 
"This is an example on selecting elements conditionally that belong to a cluster and finding its centroid."

I may have changed some things already - I just want to make sure things are clear to the audience. 
@lalith1403

Reply:
See changes sentence - if that works ? 🚀

Reply:
Thank you for the updation. Yes it conveys the meaning.